lib/mina/local_helpers.rb

```diff
@@ -64,16 +61,18 @@ def invoke(script, this)
         code = "#{script}"
 
         # Certain environments can't do :pretty mode.
-        term_mode = :exec  if term_mode == :pretty && !pretty_supported?
+        term_mode = :exec if term_mode == :pretty && !pretty_supported?
 
         case term_mode
         when :pretty
+          require 'shellwords'
+          code = Shellwords.escape(code)
           this.pretty_system(code)
         when :exec
-          exec code
+          Kernel.exec code
         else
-          system code
-          $?.to_i
+          Kernel.system code
+          $?.exitstatus
```

Comment:
Here `$?` is a `Process::Status` instance that represents a 16-bit integer that ruby got from the system. `Process::Status#to_i` returns the whole 16-bit integer and have no meaning by itself. It coincidently is 0 when the command exits with status 0, but when it exits with anything else, it's not usable, so I changed it to use `Process::Status#exitstatus` which returns the least significant 8-bits, which is the process exit status.